include/usb/class/usb_audio.h

```diff
@@ -10,11 +10,277 @@
  * @file
  * @brief USB Audio Device Class public header
  *
- * Header follows Device Class Definition for Audio Class
- * Version 1.0 document (audio10.pdf).
+ * Header follows below documentation:
+ * - USB Device Class Definition for Audio Devices (audio10.pdf)
+ *
+ * Additional documentation considered a part of USB Audio v1.0:
+ * - USB Device Class Definition for Audio Data Formats (frmts10.pdf)
+ * - USB Device Class Definition for Terminal Types (termt10.pdf)
  */
 
 #ifndef ZEPHYR_INCLUDE_USB_CLASS_AUDIO_H_
 #define ZEPHYR_INCLUDE_USB_CLASS_AUDIO_H_
 
+#include <usb/usb_common.h>
+#include <device.h>
+#include <net/buf.h>
+#include <sys/util.h>
+
+/** Audio Interface Subclass Codes
+ * Refer to Table A-2 from audio10.pdf
+ */
+enum usb_audio_int_subclass_codes {
+	USB_AUDIO_SUBCLASS_UNDEFINED	= 0x00,
+	USB_AUDIO_AUDIOCONTROL		= 0x01,
+	USB_AUDIO_AUDIOSTREAMING	= 0x02,
+	USB_AUDIO_MIDISTREAMING		= 0x03
+};
+
+/** Audio Class-Specific AC Interface Descriptor Subtypes
+ * Refer to Table A-5 from audio10.pdf
+ */
+enum usb_audio_cs_ac_int_desc_subtypes {
+	USB_AUDIO_AC_DESCRIPTOR_UNDEFINED	= 0x00,
+	USB_AUDIO_HEADER			= 0x01,
+	USB_AUDIO_INPUT_TERMINAL		= 0x02,
+	USB_AUDIO_OUTPUT_TERMINAL		= 0x03,
+	USB_AUDIO_MIXER_UNIT			= 0x04,
+	USB_AUDIO_SELECTOR_UNIT			= 0x05,
+	USB_AUDIO_FEATURE_UNIT			= 0x06,
+	USB_AUDIO_PROCESSING_UNIT		= 0x07,
+	USB_AUDIO_EXTENSION_UNIT		= 0x08
+};
+
+/** Audio Class-Specific AS Interface Descriptor Subtypes
+ * Refer to Table A-6 from audio10.pdf
+ */
+enum usb_audio_cs_as_int_desc_subtypes {
+	USB_AUDIO_AS_DESCRIPTOR_UNDEFINED	= 0x00,
+	USB_AUDIO_AS_GENERAL			= 0x01,
+	USB_AUDIO_FORMAT_TYPE			= 0x02,
+	USB_AUDIO_FORMAT_SPECIFIC		= 0x03
+};
+
+/** Audio Class-Specific Request Codes
+ * Refer to Table A-9 from audio10.pdf
+ */
+enum usb_audio_cs_req_codes {
+	USB_AUDIO_REQUEST_CODE_UNDEFINED	= 0x00,
+	USB_AUDIO_SET_CUR			= 0x01,
+	USB_AUDIO_GET_CUR			= 0x81,
+	USB_AUDIO_SET_MIN			= 0x02,
+	USB_AUDIO_GET_MIN			= 0x82,
+	USB_AUDIO_SET_MAX			= 0x03,
+	USB_AUDIO_GET_MAX			= 0x83,
+	USB_AUDIO_SET_RES			= 0x04,
+	USB_AUDIO_GET_RES			= 0x84,
+	USB_AUDIO_SET_MEM			= 0x05,
+	USB_AUDIO_GET_MEM			= 0x85,
+	USB_AUDIO_GET_STAT			= 0xFF
+};
+
+/** Feature Unit Control Selectors
+ * Refer to Table A-11 from audio10.pdf
+ */
+enum usb_audio_feature_unit_control_selectors {
```

Comment:
you have to find shorter name for it and probably other enums here, maybe `usb_audio_fucs`?

Reply:
Hmm. Wanted to underline that comes 1:1 from the spec but as there is a comment about this its worth to be shorten.